subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultArtifactTransforms.java

```diff
@@ -281,19 +281,25 @@ public void visitArtifact(AttributeContainer variant, ResolvableArtifact artifac
                 return;
             }
 
+            ResolvedArtifact sourceArtifact = artifact.toPublicView();
             List<File> transformedFiles = operation.result;
-
             TaskDependency buildDependencies = ((Buildable) artifact).getBuildDependencies();
+
             for (File output : transformedFiles) {
-                ResolvedArtifact sourceArtifact = artifact.toPublicView();
-                ComponentArtifactIdentifier newId = new ComponentFileArtifactIdentifier(sourceArtifact.getId().getComponentIdentifier(), output.getName());
-                String extension = Files.getFileExtension(output.getName());
-                IvyArtifactName artifactName = new DefaultIvyArtifactName(output.getName(), extension, extension);
+                IvyArtifactName artifactName = getArtifactName(sourceArtifact, output);
+                ComponentArtifactIdentifier newId = new ComponentFileArtifactIdentifier(sourceArtifact.getId().getComponentIdentifier(), artifactName.toString());
                 DefaultResolvedArtifact resolvedArtifact = new DefaultResolvedArtifact(sourceArtifact.getModuleVersion().getId(), artifactName, newId, buildDependencies, output);
                 visitor.visitArtifact(target, resolvedArtifact);
             }
         }
 
+        private IvyArtifactName getArtifactName(ResolvedArtifact sourceArtifact, File output) {
```

Comment:
I am never sure about our naming conventions (and maybe we don't have any for this):

Should this kind of method be named `toArtifactName()` as it is transforming the information into a new object rather than just accessing an existing object?

Reply:
We don't have any conventions around this I think. Your reasoning makes sense to me though. I'll adjust it.